sdk/core/azure-json-gson/src/main/java/com/azure/json/gson/GsonJsonReader.java

```diff
@@ -88,17 +91,25 @@ static JsonReader fromReader(Reader json, JsonOptions options) {
 
     private GsonJsonReader(Reader reader, boolean resetSupported, byte[] jsonBytes, String jsonString,
         JsonOptions options) {
-        this(reader, resetSupported, jsonBytes, jsonString, options.isNonNumericNumbersSupported());
+        this(reader, resetSupported, jsonBytes, jsonString, options.isNonNumericNumbersSupported(),
+            options.isAllowComments(), options.isAllowTrailingCommas(), options.isAllowUnescapedControlCharacters());
     }
 
     private GsonJsonReader(Reader reader, boolean resetSupported, byte[] jsonBytes, String jsonString,
-        boolean nonNumericNumbersSupported) {
+        boolean nonNumericNumbersSupported, boolean allowComments, boolean allowTrailingCommas,
```

Comment:
Are there tests for these? Or am I missing them

Reply:
I've added tests for comments as it's the most common case:

https://github.com/Azure/azure-sdk-for-java/pull/35698/files#diff-5699bbbfb83540e5e58f85c70107a344477fdcb6efe75f51250a91ea66bdc5b4R833

I can add more tests for trailing commas, not too sure what would be the best way to handled unescaped control characters.